packages/amplify_api_plugin_interface/lib/src/GraphQL/GraphQLRequest.dart

```diff
@@ -13,14 +13,22 @@
  * permissions and limitations under the License.
  */
 
-import '../UUID.dart';
+import 'package:amplify_datastore_plugin_interface/amplify_datastore_plugin_interface.dart';
+
+import '../UUID.dart' as API_UUID;
 
 class GraphQLRequest<T> {
   String document;
   Map<String, dynamic> variables;
-  String cancelToken = UUID.getUUID();
+  String cancelToken = API_UUID.UUID.getUUID();
+  String? decodePath;
```

Comment:
Something to think about is that `decodePath` and `modelType` are public-facing additions to the API. Even though they are designed to be supplied by the query helpers and users should not have to think about them, it is entirely possible for a user to manually populate a request and supply these without the helper methods. In that situation, we should at least understand how the code responds and aim to supply helpful error messages even if this is not something we encourage users to do.

No change needed in this file (although documenting these params is something to think about in the future). This is just something to consider during decoding.